onnxruntime/core/providers/webnn/builders/impl/gru_op_builder.cc

```diff
@@ -67,17 +73,13 @@ Status GruOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder, const No
     options.set("initialHiddenState", model_builder.GetOperand(input_defs[5]->Name()));
   }
 
-  uint32_t linear_before_reset = helper.Get("linear_before_reset ", 0);
-  if (linear_before_reset == 0u) {
-    options.set("resetAfter", false);
-  }
+  bool linear_before_reset = helper.Get("linear_before_reset ", 0) == 0u;
```

Comment:
```suggestion
  bool linear_before_reset = !!helper.Get("linear_before_reset ", 0);
```
Otherwise the condition appears backwards (false is true).